### PR TITLE
Remove pickles from accounting celery tasks

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

[JIRA Ticket](https://dimagi-dev.atlassian.net/browse/SAAS-11385)

#### Review commit by commit. I tried to leave helpful messages in each commit, but will give a brief summary here as well.

This PR is part of a series of PRs aimed at removing our use of pickling in celery tasks due to a lack of support that is blocking our ability to upgrade celery.

This PR removes the `serializer='pickle'` lines from tasks in the `accounting` app, which means task will instead rely on the default, json serialization. Each commit addresses one task, or a group of tasks if the change is simple enough. Here's a summary of how the various commits replace pickling with json based on what arguments the task expects:
- every argument is already json serializable ([e1411cb](https://github.com/dimagi/commcare-hq/commit/90ec203c6c0a04bdd5d296d67556f6e037cd7f2d))
- the parameter is only used in tests ([90ec203](https://github.com/dimagi/commcare-hq/pull/30277/commits/e1411cb050436a9f70dc9e23f4f4ae13a104a6d3))
- instead of passing db object, pass the id of the db object ([cd450af](https://github.com/dimagi/commcare-hq/commit/cd450af934d8cac0ce6d99d4feaf7c45293a391b) and [b0fdbb6](https://github.com/dimagi/commcare-hq/commit/b0fdbb659640731823cbccc99270cb6534a2087b))
- use `simplejson` to serialize a python object, in this case `Decimal` ([e0a8968](https://github.com/dimagi/commcare-hq/commit/e0a896812ba17fc072ae662fa5c122e724ccb8f0))

### A sidenote on testing
In the `test_task_utils.py`, I ended up relying heavily on mocking, potentially too heavily. I'm curious to hear feedback from both the robustness and readability of tests because of that reliance on mocking, specifically using a context manager to setup state for tests.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Automated tests were added to cover existing code as well as some of the refactor
### QA Plan

[Exploratory QA ticket](https://dimagi-dev.atlassian.net/browse/QA-3447)
### Safety story

I do think this requires more testing and will look into how I can go about that.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by biyeun — APPROVED** [review]
excellent cleanup and tests!!

**Comment by biyeun** [comment]
@gherceg test failure looks real

**Review by millerdev — COMMENTED** [review]
Thanks for adding tests!

File: corehq/apps/domain/utils.py [hunk 1]
```diff
@@ -113,3 +115,14 @@ def clear_domain_names(*domain_names):
     for domain_names in domain_names:
         for domain in iter_all_domains_and_deleted_domains_with_name(domain_names):
             domain.delete()
+
+
+def get_serializable_wire_invoice_item_from_request(request):
+    general_credit = Decimal(request.POST.get('general_credit', 0))
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/domain/utils.py:121]: Yeah this is the same behavior as before, but happy to make an improvement. Is it reasonable to say if someone passes a non-number value for `general_credit`, we want the exception to be raised? I'm still unsure of when to catch and handle errors vs when to allow them to be raised. I could go either way on this one.

File: corehq/apps/accounting/payment_handlers.py [hunk 2]
```diff
@@ -464,9 +464,9 @@ def _handle_card_errors(invoice, e):
...
 
     @staticmethod
-    def _handle_email_failure(payment_record):
+    def _handle_email_failure(payment_record_id):
         log_accounting_error(
             "[Autopay] During an automatic payment, sending a payment receipt failed"
-            " for Payment Record: {payment_record}. Everything else succeeded"
-            .format(payment_record=payment_record.id)
+            " for Payment Record: {payment_record_id}. Everything else succeeded"
+            .format(payment_record_id=payment_record_id)
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/accounting/payment_handlers.py]: Updated this one as well as the other 2 similar error messages to use f-strings.

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Review by biyeun — APPROVED** [review]
looks good. I think my big worry with the tests is that if the Invoice or Subscription models change, it might not be picked up with the mocked objects.

**Comment by gherceg** [comment]
I realized I wasn't actually handling the `amount` variable when passing it into the `create_wire_credits_invoice` on the factory. Rather than clutter that area of code more, I decided to move the serialization of amount and general credit to the factory right before kicking off the celery task. 

I tested this locally by creating a factory and calling `create_wire_credits_invoice` in a shell with a celery worker running and it was successful.

**Review by millerdev — APPROVED** [review]

**Pull request merged**